### PR TITLE
Multiple config tags support when plugin faces runtime error.

### PR DESCRIPTION
**Pull request by mashhurs**
# _Refer to this [PR](https://github.com/logstash-plugins/logstash-filter-kv/pull/100)_

### CODE REVIEW

File: spec/filters/kv_spec.rb [hunk 1]
```diff
@@ -1147,11 +1144,21 @@
       plugin.filter(event)
     end
     context 'when a custom tag is defined' do
-      let(:options) { super().merge("tag_on_failure" => "KV-ERROR")}
+      let(:options) { super().merge("tag_on_failure" => ["KV-ERROR"])}
```
Comment (yaauie) [inline spec/filters/kv_spec.rb:1147]: We want to keep this spec in place, to validate that old-style configs with a single string being given still work without being broken. If this spec fails, then our implementation breaks the user experience and we need to find an implementation that doesn't do so.

```suggestion
      let(:options) { super().merge("tag_on_failure" => "KV-ERROR")}
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## 4.7.0
+- Allow attaching multiple tags on failure. Changed type of `tag_on_failure` from string to array [#92](https://github.com/logstash-plugins/logstash-filter-kv/issues/92)
```
Comment (yaauie) [inline CHANGELOG.md:2]: A changelog indicating that we are changing types can be scary, and can lead people to believe that this is a breaking change when we are intending to implement this in a backward-compatible way. Let's rephrase it to indicate that this change allows a user to specify multiple tags.

```suggestion
- Allow attaching multiple tags on failure. The `tag_on_failure` option now also supports an array of strings [#92](https://github.com/logstash-plugins/logstash-filter-kv/issues/92)
```

File: spec/filters/kv_spec.rb [hunk 3]
```diff
@@ -1147,11 +1144,21 @@
...
+        expect(event.get('tags')).to_not include('_kv_filter_error')
+      end
+    end
+    context 'when multiple custom tags are defined' do
+      let(:options) { super().merge("tag_on_failure" => ["_kv_filter_error", "_kv_filter_pattern"])}
+      it 'tags the event with the custom tag' do
+        plugin.filter(event)
+        expect(event.get('tags')).to_not be_nil
+        expect(event.get('tags')).to include('_kv_filter_error')
+        expect(event.get('tags')).to include('_kv_filter_pattern')
```
Comment (yaauie) [inline spec/filters/kv_spec.rb:1161]: Lets include two tags that do _not_ overlap the default, so that we can specify that the result doesn't include the default when an array is presented:

```suggestion
      let(:options) { super().merge("tag_on_failure" => ["kv_FAIL_one", "_kv_fail_TWO"])}
      it 'tags the event with the custom tags' do
        plugin.filter(event)
        expect(event.get('tags')).to_not be_nil
        expect(event.get('tags')).to include('kv_FAIL_one')
        expect(event.get('tags')).to include('_kv_fail_TWO')
        expect(event.get('tags')).to_not include('_kv_filter_error')
```

**Review by yaauie — CHANGES_REQUESTED** [review]
It looks like CI wasn't kicked off, likely because your e-mail wasn't in the contributors list at the time you pushed. I've run it locally against 7.17.1 and 8.0.1 and things are looking on track. I've added two suggested changes.

If CI still doesn't kick-off on your next push, we can chase that down separately.

File: spec/filters/kv_spec.rb (hunk 1 above)
Comment (mashhurs) [reply spec/filters/kv_spec.rb]: Makes sense, thanks. Updated.

**Review by yaauie — APPROVED** [review]
LGTM 👍🏼
